### PR TITLE
replace AttributeTransfrom with JsonSchemaTransfrom

### PR DESCRIPTION
**Pull request by OsmondJiang**
This is just to replace AttributeTransform with Json Schema Transform, doesn't change SDP current pipeline any logic. So it's just a refactoring.

### CODE REVIEW

File: src/docfx/lib/schema/JsonSchemaTransform.cs [hunk 1]
```diff
@@ -3,36 +3,44 @@
...
-        private static void Transform(JsonSchemaContext jsonSchemaContext, Document file, Context context, JsonSchema schema, JToken token, List<Error> errors, Action<Document> buildChild)
+        public static (List<Error> errors, Dictionary<string, Lazy<JValue>>) TransformXref(Document file, Context context, JsonSchema schema, JToken token)
+        {
+            var errors = new List<Error>();
+            var extensions = new Dictionary<string, Lazy<JValue>>();
+            Travel(new JsonSchemaContext(schema), schema, token, (jsonSchema, scalar) => extensions[scalar.Path] = new Lazy<JValue>(() => TransformScalar(file, context, jsonSchema, scalar, errors, buildChild: null), LazyThreadSafetyMode.PublicationOnly));
+            return (errors, extensions);
+        }
+
+        private static void Travel(JsonSchemaContext jsonSchemaContext, JsonSchema schema, JToken token, Action<JsonSchema, JValue> transformScalar)
```
Comment (live1206) [inline src/docfx/lib/schema/JsonSchemaTransform.cs]: Travel -> Traverse

File: src/docfx/build/xref/XrefMap.cs [hunk 2]
```diff
@@ -468,26 +468,27 @@ private static (List<Error> errors, IReadOnlyList<InternalXrefSpec> specs) LoadS
...
+                throw Errors.SchemaNotFound(file.Mime).ToException();
+            }
 
+            var (schemaErrors, extensionData) = JsonSchemaTransform.TransformXref(file, context, jsonSchema, obj);
             errors.AddRange(schemaErrors);
 
             var extensionDataByUid = new Dictionary<string, (bool isRoot, Dictionary<string, Lazy<JValue>> properties)>();
 
-            foreach (var (uid, value) in uidToJsonPath)
+            foreach (var (uid, jsnoPath) in uidToJsonPath)
```
Comment (mingwli) [inline src/docfx/build/xref/XrefMap.cs]: >jsnoPath [](start = 31, length = 8)

typo

**Comment by OsmondJiang** [comment]
image

We need re-create this PR in docfx repo? not a fork repo?

**Comment by yufeih** [comment]
> image
> 
> We need re-create this PR in docfx repo? not a fork repo?

I am working on enabling secrets for team members from forked repos, hopefully it could be done by today. Before that, yes, it needs to be a PR from docfx repo for test to work.

**Comment by OsmondJiang** [comment]
> > image
> > We need re-create this PR in docfx repo? not a fork repo?
> 
> I am working on enabling secrets for team members from forked repos, hopefully it could be done by today. Before that, yes, it needs to be a PR from docfx repo for test to work.

Great to know, I can wait.

**Pull request merged**